src/components/app/details/appDetails/utils.tsx

```diff
@@ -213,27 +213,56 @@ export function getCalendarValue(startDateStr: string, endDateStr: string): stri
     return str;
 }
 
-export function getIframeSrc(appId: string | number, envId: string | number, environmentName: string, chartName: ChartTypes, newPodHash: string, calendarInputs, tab: AppMetricsTabType, isLegendRequired: boolean, statusCode?: string):string {
+export function isK8sVersion115AndBelow(k8sVersion: string): boolean {
+    let target = [1, 15, 0];
+    let version: string[] = []
+    try {
+        version = (k8sVersion.split("v")[1]).split(".");
+    } catch (error) {
+        
+    }
+    let versionNum: number[] = version.map(item => Number(item));
+    for (let i = 0; i < target.length; i++) {
+        if (versionNum[i] <= target[i]) {
+            return true;
+        }
+    }
+    return false;
+}
+
+export function getIframeSrc(appId: string | number, envId: string | number, environmentName: string, chartName: ChartTypes, newPodHash: string, calendarInputs, tab: AppMetricsTabType, isLegendRequired: boolean, k8sVersion: string, statusCode?: string): string {
     let rootUrl = process.env.REACT_APP_ORCHESTRATOR_ROOT.replace('/orchestrator', '');
-    // rootUrl = 'http://demo.devtron.info:32080';
     let startTime: string = calendarInputs.startDate;
     let endTime: string = calendarInputs.endDate;
-    let url = ``;
-    if (chartName !== 'status') {
-        url = `${rootUrl}/grafana/d-solo/devtron-app-metrics-`;
+    let url = '';
+    if (isK8sVersion115AndBelow(k8sVersion) && (chartName === 'cpu' || chartName === 'ram')) {
+        url = `${rootUrl}/grafana/d/devtron-app-metrics-`;
+        if (chartName === 'cpu') url = `${url}cpu-k8s15`;
+        if (chartName === 'ram') url = `${url}memory-k8s15`;
     }
     else {
-        url = `${rootUrl}/grafana/d-solo/NnFpQOKGk/res_status_per_pod`;
+        if (chartName !== 'status') {
+            url = `${rootUrl}/grafana/d-solo/devtron-app-metrics-`;
+        }
+        else {
+            url = `${rootUrl}/grafana/d-solo/NnFpQOKGk/res_status_per_pod`;
+        }
     }
     switch (chartName) {
         case 'latency':
             url += `latency/latency`;
             break;
         case 'ram':
-            url += `memory/memory-usage`;
+            if (isK8sVersion115AndBelow(k8sVersion)) {
+                url = `${url}/memory-usage-k8s15`;
```

Comment:
in this line we have `/` in line 259, we dont have slash, also lets use standard mechanism to concatenate url  either like this or like line 259